doc/sphinx/developer_manual/contributing_to_the_code/tips_and_tricks.md

```diff
@@ -4,4 +4,79 @@ Tips and tricks
 
 ```{todo}
 Give some tick and tricks, like start copying plugins which already so almost what you want.
-```
\ No newline at end of file
+```
+
+# Tricks to Run Tests
+
+## Trick 1: Run All Tests
+
+This command would compile and run all the tests.
+
+    make test
+
+Not so much information is given for the failing tests, so the next step is needed for more information.
+
+## Trick 2: Run one test with more information
+
+This command would run one test and print more information to the screen:
+
+    ctest -V -R name_of_test
+
+This is useful to check for failing tests. At the end of the message, the differences between the generated file of the test and the standard file would be shown.
+
+```{todo}
+Give a workflow to generate new tests
+```
+
+## Trick 3: Run the unit_tests separately
+
+In trick 1, the unit_tests are grouped as one big test. Again, not so much information is given about the individual unit tests.
+Once the tests are compiled in step 1, a separate executable could be found in the build folder which provides more information regarding the unit tests:
+
+    ./build/unit_tests
+
+```{todo}
+Give a workflow to generate new unit tests
+```
+
+## Trick 4: Interact with visualization when generating new tests
+
+The tests in the WorldBuilder contain a wb file for the setup and a dat file for the mesh point. 
+For example, the app_spherical_2d test contains two files:
+
+    tests/gwb-dat/app_spherical_2d.wb
+    tests/gwb-dat/app_spherical_2d.dat
+
+And a folder:
+
+     tests/gwb-dat/app_spherical_2d
+
+
+The first few lines of the contents in the dat file are:
+
+    # This is a comment in the data
+    # file. 
+    # Now define parameters:
+    # dim = 2
+    # compositions = 7
+    # grain compositions = 2
+    # number of grains = 2
+    # r long lat d T c1 c2 c3 c4 c5
```

Comment:
hmm, this might not be the best introduction example, since this line is actually incorrect. I should fix that at some point I guess. 

For a Cartesian case replace `r long lat` with `x y z`. For a spherical case, it by default uses Cartesian coordinates. In 3D you can use `# convert spherical = true` so that the values are interpenetrated as `R long lat`

Reply:
I changed the example to the mass_conserving_slab_temperature where the grid is 2d.